app/vmui/packages/vmui/src/components/Configurators/StepConfigurator/StepConfigurator.tsx

```diff
@@ -40,12 +40,12 @@ const StepConfigurator: FC<StepConfiguratorProps> = ({ defaultStep, setStep }) =
   };
 
   useEffect(() => {
-    if (defaultStep) handleSetStep(defaultStep);
-  }, [defaultStep]);
+    if (value) handleSetStep(value);
+  }, [value]);
 
   return (
     <TextField
-      label="Step value"
+      label="Step value of seconds"
```

Comment:
I think, it would be better to show a unit in the text area directly.
Currently, there is no way to specify milliseconds as step even if VictoriaMetrics supports millisecond precision for step.
Probably, using a unit next to the value is the best option. So you immediately show user what is the current step (for example, `5s`). And also give them an option to specify whatever unit they want: `5m` or `5ms`.
The list of supported units could be the following: `ms`, `s`, `m`, `h`, `d`.

Reply:
It would be great to allow entering the step value as the usual PromQL duration - VictoriaMetrics accepts such `step` values. If the user enters the step without duration suffix, then it should be interpreted as seconds (e.g. in the same way as now).

@Loori-R , could you do this in a new pull request?

Reply:
FYI, this feature has been implemented in https://github.com/VictoriaMetrics/VictoriaMetrics/pull/3561 .